Feature/ex65 service refactor

hptrk opened the pull request:
ez a PR még mindig a többi be nem engedett branchre épül, nincs túl sok változtatás benne amúgy

hptrk committed:
#EX65 implement hibernate filters, user caching, service refactors (not ready)

hptrk committed:
#EX65 make services more efficient, implement token filtering

hptrk commented on backend/exence/src/main/java/com/exence/finance/modules/auth/service/impl/AuthServiceImpl.java:
```diff
@@ -18,40 +18,46 @@
                 .password(passwordEncoder.encode(request.getPassword()))
                 .build();
+
         User savedUser = userRepository.save(user);
```
majd javítva lesz, van rá ticket #EX-101

hptrk commented on backend/exence/src/main/java/com/exence/finance/modules/auth/service/impl/UserServiceImpl.java:
```diff
@@ -3,77 +3,92 @@
+        }
+
+        String userEmail = authentication.getName();
+        //TODO: remove log?
```
nyitott kérdés, bent marad

hptrk commented on backend/exence/src/main/java/com/exence/finance/modules/auth/service/impl/UserServiceImpl.java:
```diff
@@ -3,77 +3,92 @@
+        User user = getCurrentUser();
 
-        // TODO: validation for updating password (e.g oldpassword check)
+        // TODO: validation for updating password (e.g currentPassword check, old passwords check, etc.)
```
van rá ticket, #EX-98

hptrk commented on backend/exence/src/main/java/com/exence/finance/common/validators/PasswordValidator.java:
```diff
@@ -0,0 +1,69 @@
+        }
+
+        if (!SPECIAL_CHAR_PATTERN.matcher(password).matches()) {
+            violations.add("must contain at least one special character (" + SPECIAL_CHAR_PATTERN + ")");
```
nem ide illik, majd javítjuk ha átnézzük az összes üzenetet

hptrk commented on backend/exence/src/main/java/com/exence/finance/config/ApplicationConfig.java:
```diff
@@ -21,14 +21,15 @@ public class ApplicationConfig {
     public UserDetailsService userDetailsService() {
         return username -> userRepository.findByEmail(username)
-                .orElseThrow(() -> new UsernameNotFoundException("User not found"));
+                .orElseThrow(() -> new UsernameNotFoundException(""));
```
elkapja a global handler, lesz üzenete

ntamasa commented on backend/exence/src/main/java/com/exence/finance/modules/auth/service/impl/AuthServiceImpl.java:
```diff
@@ -23,12 +23,14 @@
 
 @Service
 @RequiredArgsConstructor
+// TODO: átnézni az egész authservicet
```
Inkább vegyél fel róla Jiraban issuet, ne commenteljuóük szét.
Most azt mondom maradjon de ilyenekről, hogy nézzük át ne nagyon legyen.

ntamasa commented on backend/exence/src/main/java/com/exence/finance/modules/auth/service/impl/AuthServiceImpl.java:
```diff
@@ -125,6 +127,20 @@ public void refreshToken(HttpServletRequest request, HttpServletResponse respons
 
+//    @Transactional
+//    @CacheEvict(value = {"currentUser", "currentUserId"}, key = "#authentication.getName()")
+//    public void logout(Authentication authentication) {
```
látom, hogy a következő commitba megcsinálod, de addig ne commenteljök ki pls, ha pont erre a commitra állok akkor keresgélhetem, hogy miért nem működik a logout, mikor azt sehol sem jelzem, vagy írjuk odda commit descriptionbe ha ki akarod commentelni, hogy "logout does not work"

ntamasa commented on backend/exence/src/main/java/com/exence/finance/modules/auth/service/impl/UserServiceImpl.java:
```diff
@@ -3,80 +3,96 @@
+        }
+
+        String userEmail = authentication.getName();
+        //TODO: remove log?
```
ezt vedd ki
kérdés todo = kérdés / poll chatre
(szerintem maradhat a log benne, majd később ha fölöslegesnek véljük kivesszük)

hptrk replied:
cachelés tesztelése miatt maradt benne, kivettem

ntamasa commented on backend/exence/src/main/java/com/exence/finance/modules/auth/service/impl/UserServiceImpl.java:
```diff
@@ -3,80 +3,96 @@
+        User user = getCurrentUser();
 
-        // TODO: validation for updating password (e.g oldpassword check)
+        // TODO: validation for updating password (e.g currentPassword check, old passwords check, etc.)
```
szintén

hptrk replied:
nem akarom kivenni, jelzi hogy ez a metódus közel sincsen készen, ez egy másik task hatóköre, ami már fel van véve jiraban

ntamasa commented on backend/exence/src/main/java/com/exence/finance/modules/auth/service/impl/UserServiceImpl.java:
```diff
@@ -3,80 +3,96 @@
-        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
-        User user = (User) authentication.getPrincipal();
-        Long userId = user.getId();
+    private User loadUserFromAuthentication(Authentication authentication) {
```
feleslegesen bonyolultnak érzem ezt a nevet, nem lenne elég egy `getCurrentUser()`?

hptrk replied:
már van olyan hogy `getCurrentUser`, az szerepel a cacheünkben, ez a `loadUserFromAuthentication` az Authentication spring securitys obejctből szedi ki a usert `getPrincipal` segítségével, ami tartalmazhat többféle típusú infót a userről. szóval fontos hogy ez legyen az elnevezése, muszáj hogy az authentication objectre utaljon

ntamasa replied:
oksa, értem

ntamasa commented on backend/exence/src/main/java/com/exence/finance/modules/category/service/impl/CategoryServiceImpl.java:
```diff
@@ -1,53 +1,49 @@
-        if (existingCategory.isPresent()) {
-            throw new CategoryAlreadyExistsException();
-        }
+        // TODO: valahogy máshogy megoldani + kérés nélkül?
```
egyelőre maradjon szerintem benne, a + kéréseses check, szerintem még mindig jobb, hogy picit lassabb minthogy enged két ugyan olyan kategóriát létrehozni

csinálj új issuet róla, de az szerinted nem lenne jó megoldás, hogy ha már a `userServiceImpl.getCurrentUser()` úgyis `User`-t ad vissza aminek van egy olyan fieldje, hogyy categories, abban nem lehetne szimplán kikeresni, hogy létezik-e valami szép gyors search-el? nincs kérés cserébe listában keres ami nem tudom mennyivel költséghatékonyabb

hptrk replied:
1: outdated a fájl, azóta már megoldottam, hogy az SQL unique constraint violationt dob, amit a `globalExceptionHandler` elkap, tehát nem kell külön checkelnünk (felesleges db kéréssel)

2: a memóriában sokkal lassabb intézni dolgokat, mindig gondolj abba bele, hogy mi van, ha a usernek 15000 kategóriája van, azt mind filterezni memóriába nem túl okés. az a leggyorsabb, ha intézünk egy insertet a DB-be, és a DB szól, hogy unique constraintet sért meg, mi ezt elkapjuk, és mindenki boldog, nem kell 15000 elemet filterezni (még a kód is szebb, nincs telehányva egyértelmű checkekkel, amiket a db is meg tud nézni, nem kell komplex üzleti logika hozzá)

ntamasa replied:
yes, azért is mondtam, hogy nem tudom mennyivel lenne költséghatékonyabb
95% esélyt adtam volna annak, hogy ez lassabb de feldobtam hátha rosszul tudom.

hptrk committed:
#EX65 refactor static data, clean up unnecessary logging

hptrk committed:
resolve merge conflicts

# Conflicts:
#	backend/exence/src/main/java/com/exence/finance/common/exception/GlobalExceptionHandler.java
